src/main/java/org/apache/commons/compress/archivers/zip/ExtraFieldUtils.java

```diff
@@ -379,11 +382,28 @@ public ZipExtraField onUnparseableExtraField(final byte[] data, final int off, f
      * </p>
      *
      * @param c the class to register
+     *
+     * todo: "at_deprecated this registration is global and shared by all consumers.", find a way to make it specific.
```

Comment:
Hi @garydgregory not a draft but we can't fix it since _now_ it is part of the public API so it is more a todo for next breaking version.
The best is likely to let the (extensions) map (lookup only, no need to type it `Map`) be wired in the zip archive directly and this to stay the default/fallback without any public registration API but this is a breaking change I don't want right now - issue occurs rarely, when user registers custom impl and [compress] is in a shared classloader (OSGi, [compress] in tomcat/lib with multiple webapps consuming it etc...).

So this will need to be revisited in a coming version but can't be right now so for me it is really a todo.

Hope it explains it clearly enough.

Reply:
IMO: Either:
- Remove the TODO, or
- Deprecate the method and make its replacement public.

Reply:
pushed fixing it